src/js/controllers/LoginController.js

```diff
@@ -1,14 +1,39 @@
 function LoginController($scope, $http) {
+    $scope.loginSubmitting = false;
+
     $scope.login = () => {
+        $scope.loginForm = {general: {}, email: {}, password: {}};
+        $scope.loginSubmitting = true;
+
+        // Do some simple input validation.
+        if (!$scope.account) {
+            console.log('no input.');
+            $scope.loginForm.general['no-input'] = true;
+            $scope.loginSubmitting = false;
+            return;
+        } else if (!$scope.account.password || $scope.account.password.length === 0) {
+            $scope.loginForm.password.required = true;
+            $scope.loginSubmitting = false;
+            return;
+        }
+
         try {
```

Comment:
Does this actually have to be in a `try` block? That is very very uncommon in Javascript...